source/common/http/utility.cc

```diff
@@ -756,6 +757,62 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
   encodeLocalReply(is_reset, std::move(prepared_local_reply));
 }
 
+bool Utility::remoteAddressIsTrustedProxy(
+    const Envoy::Network::Address::InstanceConstSharedPtr remote,
```

Comment:
See, basically we only use the smart point as parameter when we wants to change the ownership. But in this case, if we ensure this remote always be valid, then `const Envoy::Network::Address::Instance&` should be used.

If we can ensure this remote always be valid, then `OptRef<const Envoy::Network::Address::Instance>` should be used.

I think there should be the first case.

Reply:
/wait

Reply:
I think I understood you correctly. This is getting beyond my knowledge here so I wasn't quite sure. Please see the latest commit that now uses `const Network::Address:Instance&`.